packages/subscription-controller/src/types.ts

```diff
@@ -57,7 +50,16 @@ export type Subscription = {
   currentPeriodEnd: string; // ISO 8601
   status: SubscriptionStatus;
   interval: RecurringInterval;
-  paymentMethod: PaymentMethod;
+  paymentMethod: SubscriptionPaymentMethod;
+};
+
+export type SubscriptionPaymentMethod = {
+  type: PaymentType;
+  crypto?: {
+    payerAddress: string;
+    chainId: string;
```

Comment:
is chainId in hex ?  in that case it would be safe to use `Hex` as type
 
 same for `payerAddress`